### ISSUE TITLE
Adapting the model for inpainting

### ISSUE THREAD

**Issue by marcbaltes** [samar-khanna/DiffusionSat#4]
Hi, I have a question on using your code for inpainting.

I see that you finetuned Hugging Face's Stable Diffusion model using the metadata and captions. I was able to use the notebook in this repo to generate some results of my own and they look really good.

However, I saw that Hugging Face also has pipelines for inpainting using Stable Diffusion. I am new to Hugging Face but I was wondering if it would be possible to adapt your model to be used in the inpainting pipeline without having to retrain/finetune everything again. The inpainting could look something like this: (sorry for the low quality I only have access to MS paint at the moment):
![image](https://github.com/samar-khanna/DiffusionSat/assets/43048953/a358e4b4-9848-49fb-bdb8-f25aa9209300)

If you could let me know if something like this would be possible that would be great.
Thank you and thanks for releasing this code!

**Comment by samar-khanna** [comment]
Hi @marcbaltes, thanks for asking about this! I haven't personally tried HF's in-painting pipeline yet, but I think it should be possible, especially with controlnet. You can train a controlnet that accepts masks like you have shown, and then adapts a pre-trained DiffusionSat model to output the in-painted region. Let me know if you have more questions.
